app/services/github_checks_verifier.rb

```diff
@@ -15,36 +18,33 @@ def call
     exit(false)
   end
 
-  # check_name is the name of the "job" key in a workflow, or the full name if the "name" key
-  # is provided for job. Probably, the "name" key should be kept empty to keep things short
-  def initialize(ref, check_name, token, wait, workflow_name)
-    @client = Octokit::Client.new(access_token: token)
-    @repo = ENV["GITHUB_REPOSITORY"]
-    @ref = ref
-    @check_name = check_name
-    @wait = wait.to_i
-    @workflow_name = workflow_name
+  def query_check_status
+    checks = client.check_runs_for_ref(repo, ref, { :accept => "application/vnd.github.antiope-preview+json"}).check_runs
+    apply_filters(checks)
   end
 
-  def query_check_status
-    checks = @client.check_runs_for_ref(@repo, @ref, { :accept => "application/vnd.github.antiope-preview+json"}).check_runs
+  def apply_filters(checks)
     checks.reject!{ |check| check.name == workflow_name }
-    checks.select!{ |check| check.name == check_name } if !check_name.empty?
+    checks.select!{ |check| check.name == check_name } if check_name.present?
+    apply_regexp_filter(checks)
 
     checks
   end
 
+  def apply_regexp_filter(checks)
+    checks.select!{ |check| check.name[check_regexp] } if check_regexp.present?
+  end
 
   def all_checks_complete(checks)
     checks.all?{ |check| check.status == "completed" }
   end
 
   def filters_present?
-    (!check_name.nil? && !check_name.empty?) || (!check_regexp.nil? && !check_regexp.empty?)
+    check_name.present? || check_regexp.present?
   end
 
   def fail_if_requested_check_never_run(check_name, all_checks)
-    return unless !check_name.empty? && all_checks.empty?
+    return unless check_name.present? && all_checks.blank?
```

Comment:
Do we get this methods because of ActiveSupport?

Reply:
Yes. Here a really cool post about them: https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html

Reply:
cool, I use them in Rails but wasn't aware you get them for free by just adding activesupport to Gemfile 👍

Reply:
Actually, ActiveSupport has so many modules that they are not required automatically (you can though, but it's not recommended). You can cherry-pick the modules you need :)

Reply:
that's why I'm a bit puzzled. You add `activesupport` to `Gemfile` and then explicitly include the `Configurable`. However, `blank` and `present` become available too 🤔  They can't be part of configurable, can they?

Reply:
Because we are requiring `active_support/configurable`, which requires `ordered_option`, which requires `active_support/core_ext/object/blank`:
- [configurable.rb](https://github.com/rails/rails/blob/5f3ff60084ab5d5921ca3499814e4697f8350ee7/activesupport/lib/active_support/configurable.rb#L142): 
- [ordered_option.rb](https://github.com/rails/rails/blob/5f3ff60084ab5d5921ca3499814e4697f8350ee7/activesupport/lib/active_support/ordered_options.rb)